addons/sale/models/account_move_line.py

```diff
@@ -155,12 +155,13 @@ def _sale_determine_order(self):
         for move_line in self:
             if move_line.analytic_distribution:
                 distribution_json = move_line.analytic_distribution
-                sale_order = self.env['sale.order'].search([('analytic_account_id', 'in', list(int(account_id) for account_id in distribution_json.keys())),
+                account_ids = list(map(int, ''.join(list(distribution_json.keys())).split(',')))
+                sale_order = self.env['sale.order'].search([('analytic_account_id', 'in', account_ids),
                                                             ('state', '=', 'sale')], order='create_date ASC', limit=1)
```

Comment:
```suggestion
                account_ids = [int(acc_id) for key in distribution_json.keys() for acc_id in key.split(',')]
                sale_order = self.env['sale.order'].search([('analytic_account_id', 'in', account_ids),
                sale_order = self.env['sale.order'].search([
                    ('analytic_account_id', 'in', account_ids),
                    ('state', '=', 'sale')], order='create_date ASC', limit=1)
```
NEVER use map, there is almost never a good reason to use now